crates/sargon-uniffi/src/system/sargon_os/sargon_os.rs

```diff
@@ -18,10 +19,18 @@ pub struct SargonOS {
 #[uniffi::export]
 impl SargonOS {
     #[uniffi::constructor]
-    pub async fn boot(bios: Arc<Bios>) -> Arc<Self> {
-        let internal_bios: InternalBios = bios.as_ref().clone().into();
-        let internal_sargon_os =
-            InternalSargonOS::boot(Arc::new(internal_bios)).await;
+    pub async fn boot(
+        bios: Arc<Bios>,
+        interactor: Arc<dyn HostInteractor>,
```

Comment:
```diff
- interactor: Arc<dyn HostInteractor>,
- use_factor_source_interactor: Arc<dyn UseFactorSourceInteractor>
```

And later when we pass many different interactors, we can group them like we did with `Drivers` - being a collection of many Arc<dyn Driver*****>